aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp

```diff
@@ -2500,7 +2519,12 @@ void quantize_tensor_per_tensor_affine_cpu(
   AT_DISPATCH_QINT_TYPES(
       qtensor.scalar_type(), "quantize_tensor_per_tensor_affine_cpu", [&]() {
         TORCH_CHECK(
-            rtensor.is_contiguous(), "Float tensor should be contiguous");
+            rtensor.is_contiguous(rtensor.suggest_memory_format()),
```

Comment:
how do we guarantee that these functions are always passed contiguous tensors as inputs?

Reply:
It is already broken if we are not passing contiguous tensors, because kernels extract raw pointer and operate on them. This check is to ensure that the kernels get what they expect.